src/Kestrel.Core/Internal/Http2/Http2Frame.GoAway.cs

```diff
@@ -1,37 +1,38 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Buffers.Binary;
+
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2
 {
+    /* https://tools.ietf.org/html/rfc7540#section-6.8
+        +-+-------------------------------------------------------------+
+        |R|                  Last-Stream-ID (31)                        |
+        +-+-------------------------------------------------------------+
+        |                      Error Code (32)                          |
+        +---------------------------------------------------------------+
+        |                  Additional Debug Data (*)                    |
+        +---------------------------------------------------------------+
+    */
     public partial class Http2Frame
     {
+        private const int ErrorCodeOffset = 4;
+
         public int GoAwayLastStreamId
         {
-            get => (Payload[0] << 24) | (Payload[1] << 16) | (Payload[2] << 8) | Payload[3];
-            set
-            {
-                Payload[0] = (byte)((value >> 24) & 0xff);
-                Payload[1] = (byte)((value >> 16) & 0xff);
-                Payload[2] = (byte)((value >> 8) & 0xff);
-                Payload[3] = (byte)(value & 0xff);
-            }
+            get => (int)Bitshifter.ReadUInt31BigEndian(Payload);
```

Comment:
makes me wonder if we should just make the GoAwayLastStreamId uint. Stream ids are positive 31 bit integers according to spec.

Reply:
int and uint are interchangeable for 31bit. We consistently use int for stream ids throughout the code base, and there's even a feature interface for it. That's a bigger change I don't want to make here, and it's not clear that we'd gain anything.